url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/Console.java

```diff
@@ -0,0 +1,623 @@
+        } else if ("fcl".equals(from)) {
+            importer = new FclImporter();
+        } else {
+            throw new RuntimeException("[examples error] unrecognized format "
+                    + "<" + from + "> to import");
+        }
+
+        Exporter exporter;
+        if ("fll".equals(to)) {
+            exporter = new FllExporter();
+        } else if ("fld".equals(to)) {
+            exporter = new FldExporter(" ");
+        } else if ("fcl".equals(to)) {
+            exporter = new FclExporter();
+        } else if ("fis".equals(to)) {
+            exporter = new FisExporter();
+        } else if ("cpp".equals(to)) {
+            exporter = new CppExporter();
+        } else if ("java".equals(to)) {
+            exporter = new JavaExporter();
+        } else {
+            throw new RuntimeException("[examples error] unrecognized format "
+                    + "<" + from + "> to export");
+        }
+
+        List<Op.Pair<Exporter, Importer>> tests = new ArrayList<>();
+        tests.add(new Op.Pair<Exporter, Importer>(new FllExporter(), new FllImporter()));
+        tests.add(new Op.Pair<Exporter, Importer>(new FclExporter(), new FclImporter()));
+        tests.add(new Op.Pair<Exporter, Importer>(new FisExporter(), new FisImporter()));
+
+        StringBuilder errors = new StringBuilder();
+        for (int i = 0; i < examples.size(); ++i) {
+            System.out.println("Processing " + (i + 1) + "/" + examples.size() + ": " + examples.get(i));
+            try {
+                StringBuilder text = new StringBuilder();
+                String input = sourceBase + examples.get(i) + "." + from;
+                BufferedReader source = new BufferedReader(new FileReader(input));
+                String line;
+                while ((line = source.readLine()) != null) {
+                    text.append(line).append("\n");
+                }
+                source.close();
+
+                Engine engine = importer.fromString(text.toString());
+
+                for (Op.Pair<Exporter, Importer> imex : tests) {
+                    String out = imex.first.toString(engine);
+                    Engine copy = imex.second.fromString(out);
+                    String out_copy = imex.first.toString(copy);
+
+                    if (!out.equals(out_copy)) {
+                        errors.append(String.format("[imex error] different results <%s,%s> at %s.%s",
+                                imex.first.getClass().getSimpleName(),
+                                imex.first.getClass().getSimpleName(),
+                                examples.get(i), from));
+                    }
+                }
+
+                String output = targetBase + examples.get(i) + "." + to;
+                File outputFile = new File(output);
+                if (!outputFile.exists()) {
+                    outputFile.createNewFile();
+                }
+                FileWriter target = new FileWriter(outputFile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *target*. The resource is referenced at lines 553, 559, 576. The resource is closed at line 578. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *target* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).